celery/backends/redis.py

```diff
@@ -230,25 +254,23 @@ def _params_from_url(self, url, defaults):
         else:
             connparams['db'] = path
 
+        ssl_param_keys = ['ssl_ca_certs', 'ssl_certfile', 'ssl_keyfile', 
+                          'ssl_cert_reqs']
+
+        if scheme == 'redis':
+            # Check IF connparams or query string contain ssl params, if so show warning.]
+            if (any(key in connparams for key in ssl_param_keys) or
+                any(key in query for key in ssl_param_keys)):
+                logger.warning(W_REDIS_SSL_PARAMS_AND_SCHEME_MISMATCH)
```

Comment:
I think it may be preferable to raise an exception, in order to better inform the user that they are using a wrong configuration.

Reply:
Agreed. I was concerned if raising an exception would introduce a breaking change for anyone already using code with an incorrectly structured URL, however, I guess it's better to address this so if you're happy with raising an exception, that's great. Code updated.